### PR TITLE
Undertow Endpoints provide tags when defined

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Undertow Endpoints provide tags when defined
==COMMIT_MSG==

These can allow us to set up more specific rate limiting based on the definition using simple filters around registration.
Unclear if we would want to do something similar with parameter tags.

### CODE REVIEW

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java [hunk 1]
```diff
@@ -111,6 +120,12 @@ public Builder deprecated(Optional<String> value) {
             return this;
         }
 
+        @CanIgnoreReturnValue
+        public Builder deprecated(List<String> value) {
```
Comment (ferozco) [inline conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java]: s/deprecated/tags/

Comment (carterkozak) [reply conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java]: Good catch. Should take an `Iterable` as well!

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java [hunk 2]
```diff
@@ -111,6 +120,12 @@ public Builder deprecated(Optional<String> value) {
...
+            tags = ImmutableList.copyOf(Preconditions.checkNotNull(value, "tags is required"));
```
Comment (carterkozak) [inline conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java]:
```suggestion
            tags = ImmutableList.copyOf(Preconditions.checkNotNull(value, "tags are required"));
```

**Comment by carterkozak** [comment]
Split into #1144 and #1145

**Pull request closed without merging**